### PR TITLE
DROOLS-3202: [DMN Designer] Data Type dialog - Validation - Data Type name cannot be a built in Data Type

### PR DESCRIPTION
**Pull request by karreiro**
See https://issues.jboss.org/browse/DROOLS-3202

demo

### CODE REVIEW

**Comment by jomarko** [comment]
jenkins execute full downstream build

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/common/BuiltInTypeUtils.java [hunk 1]
```diff
@@ -26,6 +26,10 @@
     public static boolean isDefault(final String type) {
         return Stream
                 .of(BuiltInType.values())
-                .anyMatch(dataType -> Objects.equals(dataType.getName(), type));
+                .anyMatch(dataType -> Objects.equals(upperCase(dataType.getName()), upperCase(type)));
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/common/BuiltInTypeUtils.java]: You might like to check `BuiltInType.getNames()` as some have an alias too.

**Comment by jomarko** [comment]
jenkins execute full downstream build

**Comment by manstis** [comment]
jenkins execute full downstream build

**Comment by karreiro** [comment]
Thank you, @manstis. Changes applied :-)

**Comment by jomarko** [comment]
Ok, need more time to review as pushed new commit and FDB wars (if were present) are gone, decided to build locally, as usually.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/common/BuiltInTypeUtils.java (hunk 1 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/common/BuiltInTypeUtils.java]: Thanks @karreiro

**Comment by karreiro** [comment]
Jenkins please retest this.

**Pull request merged**